llvm/test/CodeGen/DirectX/metadata-stripping.ll

```diff
@@ -0,0 +1,39 @@
+; RUN: opt -S --dxil-prepare %s | FileCheck %s
+
+; This test tests the whitelist inside of DxilPrepare.cpp.
+; It ensures that certain metadata nodes are removed that aren't
+; in the whitelist, and that certain nodes may remain that
+; are on the whitelist.

```

Comment:
I think we prefer the term allowlist these days. Also this is a bit verbose, maybe just
```
; Test that only metadata nodes that are valid in DXIL are allowed through
```